### PR TITLE
node: minimize hardcoded service initialization

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: node/setup.go [hunk 1]
```diff
@@ -700,7 +700,7 @@ func createPEXReactorV2(
 	logger log.Logger,
 	peerManager *p2p.PeerManager,
 	router *p2p.Router,
-) (*pex.ReactorV2, error) {
+) (service.Service, error) {
```
Comment (williambanfield) [inline node/setup.go:703]: Mind explaining why we should change this return? It seems like we can just as easily use the result as a `service.Service` without losing the extra type information that having the return value be typed would give us?

File: node/node.go [hunk 2]
```diff
@@ -438,19 +429,37 @@ func makeNode(config *cfg.Config,
...
 		pexReactor:       pexReactor,
-		pexReactorV2:     pexReactorV2,
 		evidenceReactor:  evReactor,
-		evidencePool:     evPool,
-		proxyApp:         proxyApp,
 		indexerService:   indexerService,
 		eventBus:         eventBus,
-		eventSinks:       eventSinks,
+
+		rpcEnv: &rpccore.Environment{
```
Comment (williambanfield) [inline node/node.go:440]: Nice! this is much better than having the 'ConfigureRPC` call!

File: node/node.go [hunk 3]
```diff
@@ -71,16 +69,12 @@ type nodeImpl struct {
 	mempool          mempool.Mempool
 	stateSync        bool               // whether the node should state sync on startup
 	stateSyncReactor *statesync.Reactor // for hosting and restoring state sync snapshots
-	consensusState   *cs.State          // latest consensus state
 	consensusReactor *cs.Reactor        // for participating in the consensus
-	pexReactor       *pex.Reactor       // for exchanging peer addresses
-	pexReactorV2     *pex.ReactorV2     // for exchanging peer addresses
-	evidenceReactor  *evidence.Reactor
-	evidencePool     *evidence.Pool // tracking evidence
-	proxyApp         proxy.AppConns // connection to the application
```
Comment (williambanfield) [inline node/node.go:80]: I'm assuming we just use these to construct other types and are therefore not needed for storage on `node` itself?

File: node/setup.go (hunk 1 above)
Comment (tychoish) [reply node/setup.go:703]: this seemed easier, I think this is just an intermediate step, as (eventually) the constructor will move into the pex package and it'll just be a service at that point. 

We don't need (or want) the extra type information.

File: node/node.go (hunk 3 above)
Comment (tychoish) [reply node/node.go:80]: yeah, I think these types either become service.Services or are only used to construct the RPC Environment.

**Review by williambanfield — APPROVED** [review]

**Pull request merged**